### PR TITLE
Add Toast component

### PR DESCRIPTION
**Pull request by kiyopikko**
Only one toast can be shown. 

If you want to show multiple toasts, better to use https://github.com/jossmac/react-toast-notifications

[Image from Gyazo](https://gyazo.com/1eefb059868f96cbed77d4750d87daf6)

### CODE REVIEW

File: src/components/Initial.js [hunk 1]
```diff
@@ -94,9 +96,14 @@ const Initial = ({
       </Head>
       <Header />
       <div className="container">
-        <ErrorAlert>
-          {tokenBalance.errorEthToUSD && 'can not get USD balance now'}
-        </ErrorAlert>
+        <Toast
+          isShown={!!tokenBalance.errorEthToUSD}
```
Comment (daiki44) [inline src/components/Initial.js:100]: we don't need `!!`

Comment (kiyopikko) [reply src/components/Initial.js:100]: oh, `errorEthToUSD` is boolean...!

**Comment by kiyopikko** [comment]
> Also, could you support multiple toasts?

OK, I use https://github.com/jossmac/react-toast-notifications !

File: src/hooks.js [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+
+export function usePrevious(value) {
+  const ref = useRef()
+  useEffect(() => {
+    ref.current = value
+  })
+  return ref.current
+}
+
+export function useReactToast({ errors, removeError }) {
```
Comment (kiyopikko) [inline src/hooks.js:13]: add custom hook to use `react-toast-notifications`

File: src/hooks.js [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+  const { addToast, toastStack } = useToasts()
+  const prev = usePrevious({ errors, toastStack })
+
+  useEffect(() => {
+    if (
+      (!prev && errors.length) ||
+      (prev && errors.length > prev.errors.length)
+    ) {
+      addToast(errors[errors.length - 1], { appearance: 'error' })
```
Comment (kiyopikko) [inline src/hooks.js:22]: When added an item to `errors` array (from redux state), show Toast.

File: src/hooks.js [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+    }
+  }, [errors])
+
+  useEffect(() => {
+    if (prev && toastStack.length < prev.toastStack.length) {
+      const diff = _.difference(prev.toastStack, toastStack)
+      removeError(diff[0].content)
```
Comment (kiyopikko) [inline src/hooks.js:29]: When a toast disappear, remove the toast from redux `errors`

File: src/store/tokenBalanceList.js [hunk 5]
```diff
@@ -100,12 +100,12 @@ export const getETHtoUSD = () => {
...
           dispatch(setETHtoUSD(res.data.result.ethusd))
         } else {
-          dispatch(errorSetETHtoUSD(true))
+          dispatch(pushError('can not get USD balance now'))
         }
       })
       .catch(async e => {
         console.log(e)
-        dispatch(errorSetETHtoUSD(true))
+        dispatch(pushError('can not get USD balance now'))
```
Comment (kiyopikko) [inline src/store/tokenBalanceList.js:108]: In order to show error, pass error message to `pushError`.
`pushError` can be called in any store.

File: src/store/error.js [hunk 6]
```diff
@@ -0,0 +1,20 @@
...
+export const pushError = createAction('PUSH_ERROR')
+export const removeError = createAction('REMOVE_ERROR')
+
+export const errorReducer = createReducer(
+  {
+    errors: []
+  },
+  {
+    [pushError]: (state, action) => {
+      if (!state.errors.includes(action.payload)) {
```
Comment (kiyopikko) [inline src/store/error.js:12]: Ignore duplicated error message.

**Comment by kiyopikko** [comment]
[Image from Gyazo](https://gyazo.com/d50224a90f539cd1284e1605e49c4d94)

**Pull request merged**